rb/lib/selenium/webdriver/common/print_options.rb

```diff
@@ -0,0 +1,77 @@
+# <copyright file="print_options.rb" company="Selenium Committers">
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements. See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership. The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# </copyright>
+
+module Selenium
+  module WebDriver
+    # Represents options for printing a page.
+    class PrintOptions
+      DEFAULT_SCALE = 1.0
+      DEFAULT_ORIENTATION = 'portrait'.freeze
+      DEFAULT_PAGE_SIZE = {width: 21.0, height: 29.7}.freeze # A4 size in cm
+      DEFAULT_MARGINS = {top: 1.0, bottom: 1.0, left: 1.0, right: 1.0}.freeze
+
+      attr_accessor :orientation, :scale, :background, :page_ranges, :page_size, :margins
+
+      def initialize
+        @orientation = DEFAULT_ORIENTATION
+        @scale = DEFAULT_SCALE
+        @background = false
+        @page_ranges = nil
+        @page_size = DEFAULT_PAGE_SIZE
+        @margins = DEFAULT_MARGINS
+      end
+
+      # Converts the options to a hash format to be used by WebDriver.
+      #
+      # @return [Hash]
+      def to_h
+        options = {
+          orientation: @orientation,
+          scale: @scale,
+          background: @background,
+          pageRanges: @page_ranges,
+          paperWidth: @page_size[:width],
+          paperHeight: @page_size[:height],
+          marginTop: @margins[:top],
+          marginBottom: @margins[:bottom],
+          marginLeft: @margins[:left],
+          marginRight: @margins[:right]
+        }
+
+        options.compact
+      end
+
+      # Sets the page size to a predefined size.
+      #
+      # @param [Symbol] size The predefined size (:letter, :legal, :a4, :tabloid).
+      def set_page_size(size)
```

Comment:
Compared to Python PR (https://github.com/SeleniumHQ/selenium/pull/15064), this method should accept arbitrary  width/height, not only pre-defined sizes. It's also idiomatic in Ruby to call `set_smth` methods as `smth=`. Can you please update it so that the following interface is working?

```ruby
options = Selenium::WebDriver::PrintOptions.new
options.page_size = :a4
options.page_size = :letter
options.page_size = {width: 20, height: 20} # allow custom sizes
```